data/loader.py

```diff
@@ -0,0 +1,194 @@
+            self.clear_cache(uniq_id)
+        try:
+            return self._loaded_events_by_id[uniq_id]
+        except KeyError:
+            stored_event: StoredEvent = self.load_stored_event(uniq_id)
+            self._loaded_events_by_id[uniq_id] = Event(stored_event, lazy_load=self.lazy_load)
+            return self._loaded_events_by_id[uniq_id]
+
+    def load_event(self, uniq_id: str) -> Event:
+        return self._load_event(uniq_id, reload=False)
+
+    def reload_event(self, uniq_id: str) -> Event:
+        return self._load_event(uniq_id, reload=True)
+
+    @property
+    def events_by_id(self) -> dict[str, Event]:
+        if self._events_by_id is None:
+            self._events_by_id: dict[str, Event] = {}
+            for uniq_id in self.event_uniq_ids:
+                self._events_by_id[uniq_id] = self.load_event(uniq_id)
+        return self._events_by_id
+
+    @property
+    def events_sorted_by_name(self) -> list[Event]:
+        if self._events_sorted_by_name is None:
+            self._events_sorted_by_name = sorted(
+                self.events_by_id.values(), key=lambda event: event.name)
+        return self._events_sorted_by_name
+
+    @property
+    def events_with_tournaments_sorted_by_name(self) -> list[Event]:
+        if self._events_with_tournaments_sorted_by_name is None:
+            self._events_with_tournaments_sorted_by_name = [
+                event for event in self.events_sorted_by_name if event.tournaments_by_id
+            ]
+        return self._events_with_tournaments_sorted_by_name
+
+    @property
+    def passed_events(self) -> list[Event]:
+        if self._passed_events is None:
+            self._passed_events = sorted([
+                event for event in self.events_by_id.values()
+                if event.stop < time.time()
+            ], key=lambda event: (-event.stop, -event.start, event.name))
+        return self._passed_events
+
+    @property
+    def current_events(self) -> list[Event]:
+        if self._current_events is None:
+            self._current_events = sorted([
+                event for event in self.events_by_id.values()
+                if event.start < time.time() < event.stop
+            ], key=lambda event: (-event.stop, -event.start, event.name))
+        return self._current_events
+
+    @property
+    def coming_events(self) -> list[Event]:
+        if self._coming_events is None:
+            self._coming_events = sorted([
+                event for event in self.events_by_id.values()
+                if event.public and time.time() < event.start
+            ], key=lambda event: (-event.stop, -event.start, event.name))
+        return self._coming_events
+
+    @property
+    def public_events(self) -> list[Event]:
+        if self._public_events is None:
+            self._public_events = sorted([
+                event for event in self.events_by_id.values()
+                if event.public
+            ], key=lambda event: event.name)
```

Comment:
Should be a generator expression. I think `sorted()` accepts any iterable:
```suggestion
            self._public_events = sorted((
                event for event in self.events_by_id.values()
                if event.public
            ), key=lambda event: event.name)
```

Or even (formatting could be better):
```suggestion
            self._public_events = sorted(filter(lambda event: event.public, self.events_by_id.values()), key = lambda  event: event.name)
```

Or better yet:
```suggestion
	self._public_events = sorted(filter(attrgetter('public'), self.events_by_id.values(), key=attrgetter('name'))
```

Reply:
Fixed (your last suggestion was missing a parenthesis).
Will be even nicer later with cached_property.